### DISCUSSION TITLE
자바스크립트로 a 태그 구현시 작동 문제 발생

### DISCUSSION THREAD

**Discussion by yousunzoo** [KDT4-team6/Paldo#28] [Q&A]
![image](https://user-images.githubusercontent.com/102499959/218252791-fd46470d-0feb-42d0-8d7d-52e28626554f.png)

## 사용한 코드
```javascript
const userMenu = makeDOMwithProperties("li", {
    className: "header-menu-item",
    innerHTML: `<a href="./order-list.html">${userName}님 환영합니다!</a>`,
  });
  const logoutMenu = makeDOMwithProperties("li", {
    className: "header-menu-item",
  });

  const logoutButton = makeDOMwithProperties("button", {
    class: "logout-button",
    textContent: "로그아웃",
  });
  logoutMenu.append(logoutButton);
  headerMenu.append(userMenu, logoutMenu);
```

### 이슈
* 자바스크립트로 DOM을 조작할 때 a 태그에 링크를 거니까 캡쳐처럼 페이지가 이동했다고는 뜨지만 여전히 `index.html`에 머무는 이슈가 발생했습니다. (userMenu에 innerHTML이 아닌 a 태그를 밖으로 따로 빼서 append 처리해도 동일한 문제가 발생합니다!)
* 아마 scss를 javascript 안에서 조작하려고 하면 번들 처리가 되지 않는 것과 동일한 것 같습니다.

### 생각나는 해결 방안
* headerMenu를 index.html에 두개를 만들어놓고 class를 통해 display를 조작합니다.
* 그럼 페이지 로드 시에 order-list.html도 번들 처리가 될 거라 생각합니다.
* 미리 preload 시키는 방법 외에 자바스크립트로 조작하는 방법은 생각이 나질 않습니다. 다른 의견 대환영!!!

**Comment by bongsee** [comment]
멀티페이지 상태로 개발하는 과정 중에는 parcel 진입점을 *.html로 변경하면 괜찮지 않을까 생각해요

**Reply by yousunzoo** [reply]
봉석님이 말한대로 자바스크립트에서 경로를 상대경로로 진입점을 `*.html`로 변경 시에 해당 이슈는 해결되지만 개발 서버 실행 시 index.html로 바로 연결되지 않는 이슈 확인!

일단 class로 제어하고 추후에 SPA 구현 시 변동시키도록 하겠습니다.
